container/common/helpers.go

```diff
@@ -76,16 +77,36 @@ func GetSpec(cgroupPaths map[string]string, machineInfoFactory info.MachineInfoF
 	now := time.Now()
 	lowestTime := now
 	for _, cgroupPath := range cgroupPaths {
-		// The modified time of the cgroup directory changes whenever a subcontainer is created.
+		fi, err := os.Stat(cgroupPath)
+		if err == nil && fi.ModTime().Before(lowestTime) {
+			lowestTime = fi.ModTime()
+		}
+	}
+
+	for _, cgroupPath := range cgroupPaths {
```

Comment:
Why are we iterating over `cgroupPaths` again when we already doing it at line 79? Can we not merge the two loops and just iterate once? The only difference that I see between the first loop and the second is you are updating `cgroupPath` variable based on if it's v1 or v2.

Reply:
I just thought it would be cleaner in order to make it simpler to understand the flow, but I can merge them. (If you are thinking about performance, I have no idea what the go compiler will do to cases like this). Agree that the duplicated logic is kinda stupid, so guess I can fix that anyway. Thank! :+1:

Reply:
Pushed a new change now. Does that look better @harche?

Reply:
Yes, thanks @odinuge